cmd/restic/lock_test.go

```diff
@@ -168,3 +170,47 @@ func TestLockSuccessfulRefresh(t *testing.T) {
 	// unlockRepo should not crash
 	unlockRepo(lock)
 }
+
+func TestLockWaitTimeout(t *testing.T) {
+	repo, cleanup, env := openTestRepo(t, nil)
+	defer cleanup()
+
+	elock, _, err := lockRepoExclusive(context.TODO(), repo, env.gopts)
+	test.OK(t, err)
+
+	wait := time.Second
+	env.gopts.RetryLock = wait
+
+	start := time.Now()
+	lock, _, err := lockRepo(context.TODO(), repo, env.gopts)
+	duration := time.Since(start)
+
+	test.Assert(t, err != nil,
+		"create normal lock with exclusively locked repo didn't return an error")
+	test.Assert(t, strings.Contains(err.Error(), "repository is already locked exclusively"),
+		"create normal lock with exclusively locked repo didn't return the correct error")
+	test.Assert(t, duration >= wait,
+		"create normal lock with exclusively locked repo didn't wait for the specified timeout")
+
+	test.OK(t, lock.Unlock())
+	test.OK(t, elock.Unlock())
+}
+
+func TestLockWaitSuccess(t *testing.T) {
+	repo, cleanup, env := openTestRepo(t, nil)
+	defer cleanup()
+
+	elock, _, err := lockRepoExclusive(context.TODO(), repo, env.gopts)
+	test.OK(t, err)
+
+	wait := time.Second
+	time.AfterFunc(wait/2, func() {
+		test.OK(t, elock.Unlock())
+	})
+
+	env.gopts.RetryLock = wait
+	lock, _, err := lockRepo(context.TODO(), repo, env.gopts)
```

Comment:
That will still sleep for 5 seconds, won't it?

Reply:
Yes indeed, I reduced the tests duration for the retry lock feature to 0.6s.

```
=== RUN   TestLockWaitTimeout
    testing.go:32: using low-security KDF parameters for test
    config.go:74: disabling check of the chunker polynomial
    testing.go:32: using low-security KDF parameters for test
    config.go:74: disabling check of the chunker polynomial
    lock.go:102: setting lock timeout to 0s
    integration_test.go:56: repository initialized at /tmp/restic-test-3574893611/repo
--- PASS: TestLockWaitTimeout (0.25s)
=== RUN   TestLockWaitCancel
    testing.go:32: using low-security KDF parameters for test
    config.go:74: disabling check of the chunker polynomial
    testing.go:32: using low-security KDF parameters for test
    config.go:74: disabling check of the chunker polynomial
    lock.go:102: setting lock timeout to 0s
    integration_test.go:56: repository initialized at /tmp/restic-test-1146899562/repo
--- PASS: TestLockWaitCancel (0.18s)
=== RUN   TestLockWaitSuccess
    testing.go:32: using low-security KDF parameters for test
    config.go:74: disabling check of the chunker polynomial
    testing.go:32: using low-security KDF parameters for test
    config.go:74: disabling check of the chunker polynomial
    lock.go:102: setting lock timeout to 0s
    integration_test.go:56: repository initialized at /tmp/restic-test-883179798/repo
--- PASS: TestLockWaitSuccess (0.19s)
PASS
ok  	github.com/restic/restic/cmd/restic	0.628s
```